### ISSUE TITLE
Scope variables do not accept a value containing a ':'

### ISSUE THREAD

**Issue by menuet** [conan-io/conan#542]
Hi,
I modified the Boost package recipe so that I can specify an alternate url for downloading the source zip file. I used a scope variable for this purpose. However it seems that conan does not allow scope variables to contain a ':'.

When issuing this command:
conan install Boost/1.61.0@epr-eliaskousk/stable --build -s os=Windows -s arch=x86_64 -s compiler="Visual Studio" -s compiler.version=14 -s compiler.runtime=MDd -s build_type=Debug -sc ALL:source_url=http://sourceforge.net/projects/boost/files/boost

Here is the error message I get:
ERROR: Bad scope ALL:source_url=http://sourceforge.net/projects/boost/files/boost

I suppose there is an issue because the parsing treats the ':' as a special separator?...

Would it be possible to fix this or is it supposed to be like that?
(I can workaround the issue by replacing the character ':' in my string value by yet-another special character)

thanks

**Comment by memsharded** [comment]
Hi @menuet !

First thing, just to be sure: the alternate download url retrieves different source code, or exactly the same source code? Scopes only make sense if the source code is exactly the same, otherwise, the package would be different, and you need an ``option`` to model that.

About the special character, it is true, conan use ``:`` to delimit/parse packages, e.g. the options can be specified as: ``-o Boost:shared=True``. Same for scopes, you can see the parsing code here: https://github.com/conan-io/conan/blob/develop/conans/model/scope.py#L57

It doesn't seem simple to change that, an even if we changed that, the problem just would be moved to a different special character. So I suggest keeping ``:`` in conan syntax, and if you can workaround by using a different character in your string, that would be great. Thanks for your feedback!

**Comment by menuet** [comment]
Thanks for your quick answer.
Yes, the source package is exactly the same source package than the one in sourceforge.net, except that it is local (and then very fast and not depending on internet). This is why I used a scope variable. Actually, I first used an option and then realized that the option was part of the identity of the package, which I wanted to avoid.
As a workaround, I already use the character ',' in place of ':' in my url, but I think that conan's user may encounter the same issue when using scopes (also, it is not consistent with options, which do not have the limitation of not allowing ':').
In the python code, couldn't it be possible to split first on the character '=' and then to split only the left part on the character ':'?
